receiver/prometheusreceiver/config.go

```diff
@@ -107,21 +118,43 @@ func (cfg *Config) Unmarshal(componentParser *configparser.Parser) error {
 
 	err := componentParser.UnmarshalExact(cfg)
 	if err != nil {
-		return fmt.Errorf("prometheus receiver failed to parse config: %s", err)
+		return fmt.Errorf("prometheus receiver failed to parse config: %w", err)
 	}
 
-	// Unmarshal prometheus's config values. Since prometheus uses `yaml` tags, so use `yaml`.
-	promCfg, err := componentParser.Sub(prometheusConfigKey)
-	if err != nil || len(promCfg.ToStringMap()) == 0 {
-		return err
+	var yamlBlob []byte
+
+	// Ensure that only 1 of "config_file" or the configuration exist, but not both.
+	cfgStringMap, cerr := configMapForNonFileRead(componentParser)
+	if cfg.File != "" && len(cfgStringMap) != 0 {
+		return errors.New(`prometheus receiver: either "config_file" or the Prometheus configuration can be set but not both`)
```

Comment:
Is this case tested?

Reply:
It is now tested, please take a look :-)